pkg/queue/manager.go

```diff
@@ -265,6 +265,31 @@ func (m *Manager) AddOrUpdateWorkload(w *kueue.QueuedWorkload) bool {
 	return m.addOrUpdateWorkload(w)
 }
 
+// AddWorkloadIfNotExist adds workload to the corresponding queue.
+// Returns whether the queue existed.
+func (m *Manager) AddWorkloadIfNotExist(w *workload.Info) bool {
```

Comment:
What if you add a boolean parameter to `RequeueWorkload` instead?

Otherwise there is too much repeated code.

And maybe you can do the same in the `ClusterQueue` interface.

something like: `RequeueIfNotPresent(*workload.Info, bool)`

Reply:
Not only the `ClusterQueue` but also the queue.
```go
func (m *Manager) addWorkloadIfNotExist(w *workload.Info) bool {
	qKey := queueKeyForWorkload(w.Obj)
	q := m.queues[qKey]
	if q == nil {
		return false
	}
	q.AddIfNotPresent(w)   ---> here
	cq := m.clusterQueues[q.ClusterQueue]
	if cq == nil {
		return false
	}
	if cq.PushIfNotPresent(w) { ---> here
		m.cond.Broadcast()
	}
	return true
}
```

Reply:
Queue might not need the distinction. At least not for now.

Reply:
Maybe we also need to distinguish the queue.AddOrUpdate and queue.AddIfNotPresent here?
It also can be updated if we received an update event..

Reply:
IIUC. Your suggestion is to add  a boolean parameter in`RequeueWorkload`? And call AddOrUpdate or queue.AddIfNotPresent based on the bool? Right?

Reply:
Updated. Please take a look.